napari/_vispy/layers/base.py

```diff
@@ -122,6 +125,33 @@ def _on_blending_change(self):
         self.node.set_gl_state(**blending_kwargs)
         self.node.update()
 
+    def _on_overlays_change(self):
+        # avoid circular import; TODO: fix?
+        from napari._vispy.utils.visual import create_vispy_overlay
+
+        overlay_models = self.layer._overlays.values()
+
+        for overlay in overlay_models:
+            if overlay not in self.overlays:
```

Comment:
```suggestion
            if overlay in self.overlays:
                continue
```

and dedent rest of block? For readability.